Mozu.Api/MozuClient.cs

```diff
@@ -511,6 +511,57 @@ private HttpRequestMessage GetRequestMessage()
                     _headers.Add(Headers.X_VOL_APP_CLAIMS, _apiContext.AppAuthClaim);
             }
 
+
+            AddHeader(Headers.X_VOL_VERSION, Version.ApiVersion);
+            var userAgentVal = String.Format("Mozu .Net SDK/{0} (Internal Version-{1}) ", Version.ApiVersion, Assembly.GetExecutingAssembly().GetName().Version);
+            AddHeader(Headers.USER_AGENT, userAgentVal);
+            if (_apiContext != null && !String.IsNullOrEmpty(_apiContext.CorrelationId))
+                AddHeader(Headers.X_VOL_CORRELATION, _apiContext.CorrelationId);
+
+            foreach (var key in _headers.AllKeys)
+            {
+                requestMessage.Headers.Add(key, _headers[key]);
+            }
+
+            var cacheKey = GetCacheKey(requestMessage);
+            _cacheItem = CacheManager.Instance.Get<CacheItem>(cacheKey);
+            if (_cacheItem != null)
+                requestMessage.Headers.Add("If-None-Match", _cacheItem.ETag);
+
+            return requestMessage;
+        }
+
+        private async Task<HttpRequestMessage> GetRequestMessageAsync()
```

Comment:
Looks like this method is added but not used anywhere. Is this supposed to be called for _ExecuteRequestAsyn()_?

Reply:
good catch was missing a commit